### PR TITLE
move reusable functions from ContractsFetch to new fetch-contracts library

### LINKED ISSUE #10968: move ACS/tx-fetching utilities from ContractsFetch into a library
The idea of maintaining a local database built on initializing and lazily or eagerly updating from the transaction stream is generally useful, and we have a powerful set of utilities for doing this job with varying degrees of sophistication within JSON API. These are concentrated in a set of utilities for batching and structuring ACS-and-transaction-stream requests in a concurrent-access-safe way in `ContractsFetch.scala`.

We can move these utilities into a separate library for other applications. (From a discussion with @stefanobaghino-da.)

### PR DESCRIPTION
**Pull request by S11001001**
A new library, `ContractsFetch`, which represents streaming batches of contracts and transaction deltas from the gRPC API into the json-api domain model. Fixes #10968.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by stefanobaghino-da — COMMENTED** [review]
👍🏻

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala [hunk 1]
```diff
@@ -203,41 +163,12 @@ object domain {
...
-        gler: lav1.transaction_service.GetLedgerEndResponse
-    ): Option[Offset] =
-      gler.offset.flatMap(_.value.absolute).filter(_.nonEmpty).map(x => Offset(x))
-
-    def fromLedgerApi(tx: lav1.transaction.Transaction): Offset = Offset(tx.offset)
-
-    def toLedgerApi(o: Offset): lav1.ledger_offset.LedgerOffset =
-      lav1.ledger_offset.LedgerOffset(lav1.ledger_offset.LedgerOffset.Value.Absolute(unwrap(o)))
-
+  /* TODO SC Offset where?
```
Comment (akshayshirahatti-da) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala:166]: needs cleanup ?

**Review by akshayshirahatti-da — APPROVED** [review]
apart from the `TODO` in code which seems like something that needs to be cleaned up , rest looks good.
Thanks for the refactoring.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala (hunk 1 above)
Comment (S11001001) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: fbf7101752b; thanks

**Pull request merged**